app/src/main/java/agency/five/codebase/android/movieapp/ui/home/mapper/HomeScreenMapperImpl.kt

```diff
@@ -0,0 +1,42 @@
+package agency.five.codebase.android.movieapp.ui.home.mapper
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.model.Movie
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.component.MovieCategoryLabelTextViewState
+import agency.five.codebase.android.movieapp.ui.component.MovieCategoryLabelViewState
+import agency.five.codebase.android.movieapp.ui.home.HomeMovieCategoryViewState
+import agency.five.codebase.android.movieapp.ui.home.HomeMovieViewState
```

Comment:
Not sure what you mean?

Reply:
```
    override fun toHomeMovieCategoryViewState(
        movieCategories: List<MovieCategory>,
        selectedMovieCategory: MovieCategory,
        movies: List<Movie>
    ): HomeMovieCategoryViewState {
        return HomeMovieCategoryViewState(
            movieCategories.map { category ->
                MovieCategoryLabelViewState(
                    itemId = category.ordinal,
                    categoryText = MovieCategoryLabelTextViewState.LabelTextFromResource(
                        getCategoryStringResource(category)
                    ),
                    isSelected = category == selectedMovieCategory
                )
            },
            movies.map { movie ->
                HomeMovieViewState(
                    movie.id,
                    MovieViewState(movie.imageUrl, movie.isFavorite)
                )
            }
        )
    }
```
It should look like this when properly formatted. 
Doesn't break anything so you can fix this in your next homework as well, but try to fix that `reformat code` command,  because writing code without it is not sustainable.